import "tips.proto";

package SanProto;


message BroadcastInfo {
  required int32 mode_id = 1;
  required int32 passed_time = 2;
  required bytes content =3;
  optional int32 id = 4;    //广播自身的id
  optional int32 speed = 5;    //1慢 2中 3快
}

//查询广播信息
message QueryBroadcastInfoReq {

}
message QueryBroadcastInfoRes {
  required int32 status = 1;
  repeated BroadcastInfo records = 2;
  repeated ButtonTips button_tips = 3;
  optional bool chat_available = 4;
  optional int32 chat =5; //聊天tip
}


//添加广播信息
message AddBroadcastInfoReq {
  required uint64 user_id = 1;
  required int32 mode_id = 2;
  required int32 priority = 3;
  required int32 life_time = 4;
  required bytes content = 5;
}
message AddBroadcastInfoRes {
  required int32 status = 1;
}

//删除广播信息
message DeleteBroadcastInfoReq {
  repeated int32 ids = 1;
}
message DeleteBroadcastInfoRes {
  required int32 status = 1;
}

